drivers/modem/hl7800.c

```diff
@@ -0,0 +1,4900 @@
+/*
+ * Copyright (c) 2020 Laird Connectivity
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT swi_hl7800
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(modem_hl7800, CONFIG_MODEM_LOG_LEVEL);
+
+#include <zephyr/types.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <errno.h>
+#include <zephyr.h>
+#include <drivers/gpio.h>
+#include <device.h>
+#include <init.h>
+
+#include <power/power.h>
+#include <drivers/uart.h>
+
+#include <net/net_context.h>
+#include <net/net_if.h>
+#include <net/net_offload.h>
+#include <net/net_pkt.h>
+#include <net/dns_resolve.h>
+#if defined(CONFIG_NET_IPV6)
+#include "ipv6.h"
+#endif
+#if defined(CONFIG_NET_IPV4)
+#include "ipv4.h"
+#endif
+#if defined(CONFIG_NET_UDP)
+#include "udp_internal.h"
+#endif
+
+#ifdef CONFIG_MODEM_HL7800_FW_UPDATE
+#include <fs/fs.h>
+#endif
+
+#include "modem_receiver.h"
+#include <drivers/modem/hl7800.h>
+
+#define PREFIXED_SWITCH_CASE_RETURN_STRING(prefix, val)                        \
+	case prefix##_##val: {                                                 \
+		return #val;                                                   \
+	}
+
+/* Uncomment the #define below to enable a hexdump of all incoming
+ * data from the modem receiver
+ */
+/* #define HL7800_ENABLE_VERBOSE_MODEM_RECV_HEXDUMP 1 */
+
+#define HL7800_LOG_UNHANDLED_RX_MSGS 1
+
+/* Uncomment the #define(s) below to enable extra debugging */
+/* #define HL7800_RX_LOCK_LOG 1 */
+/* #define HL7800_TX_LOCK_LOG 1 */
+/* #define HL7800_IO_LOG 1 */
+
+#define HL7800_RX_LOCK_DBG_LOG(fmt, ...)                                       \
```

Comment:
Why do you need those macros? The logging subsystem already has provisions to enable and disable logging of a particular module

Reply:
These are extra debug capabilities we want separate from just enabled `DBG`.  These are needed to turn on extra info for digging into interactions in case an advanced bug is found.  Think of this level as `DBG+`.